packages/otel/src/index.ts

```diff
@@ -0,0 +1,79 @@
+/* eslint-disable no-underscore-dangle */
+import {
+  propagation,
+  context as otelContext,
+  trace,
+  Context,
+  Tracer,
+  Span,
+  SpanKind,
+} from '@opentelemetry/api';
+import { Middleware, MiddlewareContext, Logger } from 'steveo';
+
+export class OpenTelemetryMiddleware implements Middleware {
+  tracer: Tracer;
+
+  logger: Logger;
+
+  constructor(logger: Logger) {
+    this.logger = logger;
+    this.tracer = trace.getTracer('steveo');
+  }
+
+  publish(context: MiddlewareContext, next) {
+    const meta = context?.payload?._meta ?? {};
+
+    const traceHeaders = {};
+    propagation.inject(otelContext.active(), traceHeaders);
+    const updatedPayload = {
+      ...(context?.payload ?? {}),
+      _meta: {
+        ...meta,
+        traceMetadata: JSON.stringify(traceHeaders),
+      },
+    };
```

Comment:
Something future selves need to consider is that this approach is going to mess up the FIFO deduplication based on hash of the message. We may need to change how we propagate this info - ideally via the message envelope on the queues